WMF Framework/Significant Events Endpoint/SignificantEventsModels.swift

```diff
@@ -0,0 +1,616 @@
+                
+                self.outputType = untypedEvent.outputType
+                self.revId = revId
+                self.timestampString = timestampString
+                self.user = user
+                self.userId = userId
+                self.section = section
+                self.snippet = snippet
+                self.userGroups = untypedEvent.userGroups
+                self.userEditCount = untypedEvent.userEditCount
+            }
+        }
+    }
+}
+
+//MARK: Changes
+
+public extension SignificantEvents {
+    
+    struct Change {
+        
+        public struct AddedText {
+            let outputType: ChangeOutputType
+            public let sections: [String]
+            public let snippet: String?
+            public let snippetType: SnippetType
+            public let characterCount: UInt
+            
+            fileprivate init?(untypedChange: UntypedChange) {
+                guard let snippetType = untypedChange.snippetType,
+                      let characterCount = untypedChange.characterCount else {
+                    return nil
+                }
+                
+                self.outputType = untypedChange.outputType
+                self.sections = untypedChange.sections
+                self.snippet = untypedChange.snippet
+                self.snippetType = snippetType
+                self.characterCount = characterCount
+            }
+        }
+        
+        public struct DeletedText {
+            let outputType: ChangeOutputType
+            public let sections: [String]
+            public let characterCount: UInt
+            
+            fileprivate init?(untypedChange: UntypedChange) {
+                guard let characterCount = untypedChange.characterCount else {
+                    return nil
+                }
+                
+                self.outputType = untypedChange.outputType
+                self.sections = untypedChange.sections
+                self.characterCount = characterCount
+            }
+        }
+        
+        public struct NewTemplates {
+            let outputType: ChangeOutputType
+            public let sections: [String]
+            private let untypedTemplates: [[String: String]]
+            public let typedTemplates: [Template]
+            
+            fileprivate init?(untypedChange: UntypedChange) {
+                guard let untypedTemplates = untypedChange.untypedTemplates else {
+                    return nil
+                }
+                
+                var typedTemplates: [Template] = []
+                self.outputType = untypedChange.outputType
+                self.sections = untypedChange.sections
+                self.untypedTemplates = untypedTemplates
+                
+                for untypedTemplate in untypedTemplates {
+                    guard let name = untypedTemplate["name"] else {
+                        continue
+                    }
+                    let lowercaseName = name.lowercased()
```

Comment:
Is there a functional reason we're lowercasing `name` that I'm missing? If not, might be worth using one of these to skip the indirection:

```swift
if name.localizedCaseInsensitiveContains("cite")  { ... }
if name.caseInsensitiveCompare("cite") == .orderedSame { ... }
```

Reply:
Nope no reason, this is fixed now with `localizedCaseInsensitiveContains`. The names can look something like

```
"name": "cite web ",
``` 

in the endpoint, so I still needed to keep the contains logic.